### ISSUE TITLE
For POST/PUT should body be part of the cache key?

### ISSUE THREAD

**Issue by pedroreys** [shiftkey/vcr-sharp#5]
I noticed that only the `method` + `uri` is [used as the cache key](https://github.com/shiftkey/vcr-sharp/blob/master/vcr-sharp-tests/Cassette.cs#L41-L42).

While this make perfect sense for safe methods such as `GET` and `HEAD`, it can be surprising if you make more than one `POST` `PUT` request in the same session with different body contents. Is that by design? Or should the `body` be part of the cache key for those methods?

**Comment by shiftkey** [comment]
Yeah, that was a shortcut and not intended to be the whole picture. For reference, VCR uses a number of different HTTP request checks [by default](https://github.com/vcr/vcr/blob/31e2cba76c0c9a60c3de5d5ece1c87563bfeacf7/lib/vcr/request_matcher_registry.rb#L111-L130):

 - method
 - URI
 - body
 - headers
 - host and path (after parsing into a URI) 
 - querystring
 - JSON structure of body

:thumbsup: to making this more refined, based on what's present in `HttpRequestMessage`.
